authapi/tests/test_teams.py

```diff
@@ -202,6 +493,125 @@ def test_get_team(self):
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_permission_get_team_org_member(self):
+        '''Users that are members of a team's organization should be able to
+        see the team's details.'''
+        org = SeedOrganization.objects.create()
+        team = SeedTeam.objects.create(organization=org)
+        user, token = self.create_user()
+        org.users.add(user)
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+
+        url = reverse('seedteam-detail', args=(team.pk,))
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_permission_get_team_org_admin(self):
+        '''Users that have an org:admin permission for a team's organization
+        should be able to see the team details.'''
+        org1 = SeedOrganization.objects.create()
+        org2 = SeedOrganization.objects.create()
+        team1 = SeedTeam.objects.create(organization=org1)
+        team2 = SeedTeam.objects.create(organization=org2)
+        user, token = self.create_user()
+        self.add_permission(user, 'org:admin', org1.pk)
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+
+        url = reverse('seedteam-detail', args=(team2.pk,))
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+        url = reverse('seedteam-detail', args=(team1.pk,))
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_permission_get_team_org_write(self):
+        '''Users that have an org:write permission for a team's organization
+        should be able to see the team details.'''
+        org1 = SeedOrganization.objects.create()
+        org2 = SeedOrganization.objects.create()
+        team1 = SeedTeam.objects.create(organization=org1)
+        team2 = SeedTeam.objects.create(organization=org2)
+        user, token = self.create_user()
+        self.add_permission(user, 'org:write', org1.pk)
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+
+        url = reverse('seedteam-detail', args=(team2.pk,))
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+        url = reverse('seedteam-detail', args=(team1.pk,))
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_permission_get_team_admin(self):
+        '''Admin users should have read access to all teams.'''
+        org = SeedOrganization.objects.create()
+        team = SeedTeam.objects.create(organization=org)
+        _, token = self.create_admin_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+
+        url = reverse('seedteam-detail', args=(team.pk,))
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
```

Comment:
Done in https://github.com/praekelt/seed-auth-api/pull/26/commits/b07739faff0b7903be1951512dfc727c755c6078 , did for both users and teams, hopefully it doesn't make the PR to big to review.